File: backend/src/main/java/com/trio/backend/service/ConversationService.java
package com.trio.backend.service;

import com.trio.backend.dto.communication.ConversationMemberResponse;
import com.trio.backend.dto.communication.ConversationResponse;
import com.trio.backend.dto.communication.CreateConversationRequest;
import com.trio.backend.dto.communication.UpdateConversationRequest;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.UUID;

public interface ConversationService {

    ConversationResponse create(UUID workspaceId, CreateConversationRequest request);

    ConversationResponse getById(UUID workspaceId, UUID conversationId);

    Page<ConversationResponse> listUserConversations(UUID workspaceId, Pageable pageable);

    Page<ConversationResponse> listByType(UUID workspaceId, String type, Pageable pageable);

    List<ConversationResponse> listWorkspaceDefaults(UUID workspaceId);

    List<ConversationResponse> listDirectConversations(UUID workspaceId);

    ConversationResponse update(UUID workspaceId, UUID conversationId, UpdateConversationRequest request);

    void archive(UUID workspaceId, UUID conversationId);

    void delete(UUID workspaceId, UUID conversationId);

    ConversationResponse addMember(UUID workspaceId, UUID conversationId, UUID userId);

    void removeMember(UUID workspaceId, UUID conversationId, UUID userId);

    List<ConversationMemberResponse> listMembers(UUID workspaceId, UUID conversationId);

    long getUnreadCount(UUID workspaceId, UUID conversationId);
}

File: frontend/project/src/services/conversation-service.ts
import { apiClient } from '../lib/api';
import type { PageResponse } from '../types/api';
import type {
  ConversationResponse,
  CreateConversationRequest,
  UpdateConversationRequest,
  ConversationMemberResponse,
} from '../types/communication';

function base(wsId: string) {
  return `/workspaces/${wsId}/conversations`;
}

export const conversationService = {
  list: (wsId: string) =>
    apiClient.get<PageResponse<ConversationResponse>>(`${base(wsId)}`),

  getById: (wsId: string, convId: string) =>
    apiClient.get<ConversationResponse>(`${base(wsId)}/${convId}`),

  create: (wsId: string, data: CreateConversationRequest) =>
    apiClient.post<ConversationResponse>(`${base(wsId)}`, data),

  update: (wsId: string, convId: string, data: UpdateConversationRequest) =>
    apiClient.put<ConversationResponse>(`${base(wsId)}/${convId}`, data),

  archive: (wsId: string, convId: string) =>
    apiClient.post<void>(`${base(wsId)}/${convId}/archive`),

  delete: (wsId: string, convId: string) =>
    apiClient.delete<void>(`${base(wsId)}/${convId}`),

  listByType: (wsId: string, type: string) =>
    apiClient.get<PageResponse<ConversationResponse>>(`${base(wsId)}/type/${type}`),

  listWorkspaceDefaults: (wsId: string) =>
    apiClient.get<ConversationResponse[]>(`${base(wsId)}/workspace-defaults`),

  listDirect: (wsId: string) =>
    apiClient.get<ConversationResponse[]>(`${base(wsId)}/direct`),

  addMember: (wsId: string, convId: string, userId: string) =>
    apiClient.post<ConversationResponse>(`${base(wsId)}/${convId}/members/${userId}`),

  removeMember: (wsId: string, convId: string, userId: string) =>
    apiClient.delete<void>(`${base(wsId)}/${convId}/members/${userId}`),

  listMembers: (wsId: string, convId: string) =>
    apiClient.get<ConversationMemberResponse[]>(`${base(wsId)}/${convId}/members`),

  getUnreadCount: (wsId: string, convId: string) =>
    apiClient.get<number>(`${base(wsId)}/${convId}/unread-count`),
};
